src/components/Account/SignUp.tsx

```diff
@@ -1,35 +1,153 @@
+  const [last_name, setLastName] = useState("");
+  const dispatch = useDispatch();
+  const navigate = useNavigate();
+
+  const handleSubmit = (e: any) => {
+    e.preventDefault();
+
+    const newUser = {
+      username,
+      first_name,
+      last_name,
+      email,
+      password,
+    };
+
+    fetch(`http://localhost:3000/api/users`, {
+      method: "POST",
+      headers: {
+        Accept: "application/json",
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify({ user: newUser }),
+    })
+      .then((response) => response.json())
+      .then((data) => {
+        localStorage.setItem("token", data.token);
+        dispatch(login(data));
+      });
+      navigate('/profile')
+  };
   return (
     <div className="background">
-      <div className="signup-container" style={{padding:'220px'}}>
-        <form style={{ display: "flex", flexDirection: "column", width: "500px", backgroundColor:'white', margin:'auto',borderRadius:'10px',padding:'35px', boxShadow:'8px 7px 5px 5px black'}}>
-          <h1 style={{margin:'auto'}}>
+      <div className="signup-container" style={{ padding: "220px" }}>
+        <form
+          onSubmit={(e) => handleSubmit(e)}
+          style={{
+            display: "flex",
+            flexDirection: "column",
+            width: "500px",
+            backgroundColor: "white",
+            margin: "auto",
+            borderRadius: "10px",
+            padding: "35px",
+            boxShadow: "8px 7px 5px 5px black",
+          }}
+        >
+          <h1 style={{ margin: "auto" }}>
             <em>Sign Up</em>
           </h1>
-          <Form.Group className="mb-3" controlId="formBasicEmail">
-            <Form.Label>Email address</Form.Label>
-            <Form.Control type="email" placeholder="Enter email"></Form.Control>
-          </Form.Group>
-          <Form.Group className="mb-3" controlId="formBasicPassword">
-            <Form.Label>Password</Form.Label>
-            <Form.Control type="password" placeholder="Password" />
-          </Form.Group>
           <Form.Group className="mb-3" controlId="formBasicText">
             <Form.Label>First Name</Form.Label>
-            <Form.Control type="text" placeholder="First Name" />
+            <div id="icons">
+              <img
+                src={accountIcon}
+                alt="account-icon"
+                style={{ width: "20px", height: "20px", margin: "auto 5px" }}
+              ></img>
+              <Form.Control
+                type="text"
+                placeholder="First Name"
+                value={first_name}
+                onChange={(e) => setFirstName(e.target.value)}
+              />
+            </div>
           </Form.Group>
           <Form.Group className="mb-3" controlId="formBasicText">
             <Form.Label>Last Name</Form.Label>
-            <Form.Control type="text" placeholder="Last Name" />
+            <div id="icons">
```

Comment:
can we use classname here? icons indicate that there is multiple, ids should be unique